### DISCUSSION TITLE
Use native date, time and datetime-local inputs

### DISCUSSION THREAD

**Discussion by Fedik** [joomla/joomla-cms#36641] [2. Cold Features]
Hey folks, what do you think about using of native  `date`, `time` and `datetime-local` inputs? 
And as possible replacement for our lovely calendar?
The pros is the value format always in `y-m-d`, `h:i`, `y-m-dTh:i` already, and rendering with translations done by browser.
And a big pro.: it will use native mobile inputs, for mobile devices.

Example non-Latin Firefox:
![Screenshot_2022-01-11_11-40-10](https://user-images.githubusercontent.com/1568198/148919162-b0ce2386-e5ee-45a5-ae99-317bf46504ea.png)
![Screenshot_2022-01-11_11-40-28](https://user-images.githubusercontent.com/1568198/148919191-ea002ebb-adf4-448b-8662-173cb4701360.png)

I could prepare PR for one of future Joomla , maybe for 4.2
Thoughts?

Links for test:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/time
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local

**Comment by brianteeman** [comment]
> Because of the limited browser support for datetime-local, and the variations in how the inputs work, it may currently still be best to use a framework or library to present these, or to use a custom input of your own. Another option is to use separate date and time inputs, each of which is more widely supported than datetime-local.

Is this still a valid issue

**Reply by brianteeman** [reply]
and I might be wrong but I dont think it supports non-gregorian caalendars

**Reply by Fedik** [reply]
Support is okay to me, at least for `date` and `time`.
https://caniuse.com/mdn-html_elements_input_input-date
https://caniuse.com/mdn-html_elements_input_input-datetime-local

**Reply by Fedik** [reply]
Okay, not as replacement, but as 3 new fields.

**Reply by dgrammatiko** [reply]
Introducing the fields `date`, `time` and `datetime-local` should not raise any arguments. Joomla **SHOULD** support **ALL** the native HTML Form Elements OOTB.-

Replacing the calendar is another story and I'm against it although the current calendar is not the best implementation out there...

**Comment by simbus82** [comment]
Great idea! Love it!

> _And a big pro.: it will use native mobile inputs, for mobile devices._

For me this is somewhat a "mobile ux bug" (in j3 and j4 we need to uses normal keyboard), **so for me can be really proposed for 4.0.x :-)** No need to wait 4.2 or 4.1.

Joomla 4 on Android with Chrome
![image](https://user-images.githubusercontent.com/1375730/148974264-fe90bb79-59e2-49fa-a07d-3f7787dfbd78.png)

A random Jsfiddle with a date field on Android with Chrome
![image](https://user-images.githubusercontent.com/1375730/148975018-3474390e-b881-453b-9348-43b35a3a4048.png)

The Mozilla page about datetime-local on Android with Chrome
![image](https://user-images.githubusercontent.com/1375730/148975755-333bf50d-2413-4d80-8af5-a3913e81f5e9.png)

**Reply by dgrammatiko** [reply]
It will not replace the existing calendar fields. The functionality is not on par, if Joomla switches the fields then it will support only ONE calendar. The calendar needs to be rewritten using webcomponents and the temporal that will be available later this year (there's already a polyfill).
In short, the datime-local cannot and should not replace the calendar field...

**Comment by roland-d** [comment]
@Fedik If you wish to create a PR for 4.2, I am more than happy to include this. Just to be clear, this will create 3 new form fields of the types `date`, '`time`, and `datetime-local` or is it going to be 1 field where you can set the option for `date`, '`time`, or `datetime-local`?

**Reply by Fedik** [reply]
It will be 3 fields

**Reply by Fedik** [reply]
@roland-d here it is #37456

**Comment by infograf768** [comment]
What about Farsi Jalali calendar for example? 

https://stackoverflow.com/questions/62790557/change-input-type-date-to-persian-date

**Comment by roland-d** [comment]
You mean that it is not supported by the browser?

**Reply by brianteeman** [reply]
> and I might be wrong but I dont think it supports non-gregorian caalendars

native date and time are only gregorian

**Reply by roland-d** [reply]
That I understood as well but because of that we should not include it? Perhaps we can find/create some kind of fallback. The article linked has a jQuery solution for Persian dates.

**Reply by dgrammatiko** [reply]
You can add the new fields but you cannot replace the calendar with any of them

**Reply by roland-d** [reply]
Of course, the calendar comes out of the browser with these fields otherwise you need to use the `calendar` field.
